### ISSUE TITLE
linking errors

### ISSUE THREAD

**Issue by aupiff** [osidorkin/haskell-eigen#4]
When I install the `eigen` library everything seems to build properly. However, when I try to build a haskell executable that depends on the library, I get the following linking errors:

```
Linking dist/build/grath/grath ...

~/network-analysis/.cabal-sandbox/lib/x86_64-linux-ghc-7.10.1/eigen_5JOz8fRSLKXF6ueVw89YC2/libHSeigen-2.1.0-5JOz8fRSLKXF6ueVw89YC2.a(Internal.o):(.text+0xd165): undefined reference to `eigen_sparse_compress'

~/network-analysis/.cabal-sandbox/lib/x86_64-linux-ghc-7.10.1/eigen_5JOz8fRSLKXF6ueVw89YC2/libHSeigen-2.1.0-5JOz8fRSLKXF6ueVw89YC2.a(Internal.o):(.text+0xd341): undefined reference to `eigen_sparse_compress'

~/network-analysis/.cabal-sandbox/lib/x86_64-linux-ghc-7.10.1/eigen_5JOz8fRSLKXF6ueVw89YC2/libHSeigen-2.1.0-5JOz8fRSLKXF6ueVw89YC2.a(Internal.o):(.text+0xf315): undefined reference to `eigen_sparse_lowerTriangle'

~/network-analysis/.cabal-sandbox/lib/x86_64-linux-ghc-7.10.1/eigen_5JOz8fRSLKXF6ueVw89YC2/libHSeigen-2.1.0-5JOz8fRSLKXF6ueVw89YC2.a(Internal.o):(.text+0xf4f1): undefined reference to `eigen_sparse_lowerTriangle'

~/network-analysis/.cabal-sandbox/lib/x86_64-linux-ghc-7.10.1/eigen_5JOz8fRSLKXF6ueVw89YC2/libHSeigen-2.1.0-5JOz8fRSLKXF6ueVw89YC2.a(Internal.o):(.text+0xf725): undefined reference to `eigen_sparse_upperTriangle'

~/network-analysis/.cabal-sandbox/lib/x86_64-linux-ghc-7.10.1/eigen_5JOz8fRSLKXF6ueVw89YC2/libHSeigen-2.1.0-5JOz8fRSLKXF6ueVw89YC2.a(Internal.o):(.text+0xf901): undefined reference to `eigen_sparse_upperTriangle'

collect2: error: ld returned 1 exit status
```

I've created a fix here: https://github.com/aupiff/haskell-eigen/tree/link-bug-fix, but I'm not sure it's ideal as I don't know very much about haskell C bindings. Is there something obvious that I'm doing wrong?

**Comment by osidorkin** [comment]
Thank you. I removed references to unused code. Could you please check recent version. I cannot reproduce this error on my system.

**Comment by osidorkin** [comment]
Fixed in version 2.1.4.
